### PR TITLE
🔀Hotkeys

### PR DESCRIPTION
**Pull request by Cr4zyc4k3**
**What kind of update does this PR provide?** *Please check*

- [ ] new translations / updated translations / translation fixes
- [x] a new feature
- [ ] a new module
- [ ] a bugfix for an existing feature / module
- [ ] improvement of style or user experience
- [ ] other: *Please fill out*

**What is included in your update?**
- Hotkeys for buildings

### CODE REVIEW

**Review by kdev — APPROVED** [review]

File: src/modules/hotkeys/assets/commands/building.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            if (hire == null) return;
+
+            (hire as HTMLElement).click();
+        },
+        personal() {
+            const personal = document.querySelector("a[href$='/personals']");
+            //No result → Building without personal
+            if (personal == null) return;
+
+            (personal as HTMLElement).click();
```
Comment (jxn-30) [inline src/modules/hotkeys/assets/commands/building.ts:94]:
```suggestion
            const personal = document.querySelector<HTMLAnchorElement>("a[href$='/personals']");
            //No result → Building without personal
            if (personal == null) return;

            personal.click();
```

File: src/modules/hotkeys/assets/commands/building.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            const extensionBtn = document.querySelector("a[href$='/alliance']");
+            //No result → Building without enable able extension or extensionen already disabled
+            if (
+                extensionBtn == null ||
+                document.querySelectorAll("a[href*='alliance_costs']").length ==
+                    0
+            )
+                return;
+
+            (extensionBtn as HTMLElement).click();
```
Comment (jxn-30) [inline src/modules/hotkeys/assets/commands/building.ts:123]:
```suggestion
            const extensionBtn = document.querySelector<HTMLAnchorElement>("a[href$='/alliance']");
            //No result → Building without enable able extension or extensionen already enabled
            if (
                extensionBtn == null ||
                document.querySelectorAll("a[href*='alliance_costs']").length !=
                    0
            )
                return;

            (extensionBtn as HTMLElement).click();
        },
        disableSharing() {
            const extensionBtn = document.querySelector("a[href$='/alliance']");
            //No result → Building without enable able extension or extensionen already disabled
            if (
                extensionBtn == null ||
                document.querySelectorAll("a[href*='alliance_costs']").length ==
                    0
            )
                return;

            extensionBtn.click();
```

File: src/modules/hotkeys/assets/commands/building.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        },
+        toggleSharing() {
+            const extensionBtn = document.querySelector("a[href$='/alliance']");
+            //No result → Building without enable able extension
+            if (extensionBtn == null) return;
+
+            (extensionBtn as HTMLElement).click();
```
Comment (jxn-30) [inline src/modules/hotkeys/assets/commands/building.ts:130]:
```suggestion
            const extensionBtn = document.querySelector<HTMLAnchorElement>("a[href$='/alliance']");
            //No result → Building without enable able extension
            if (extensionBtn == null) return;

            extensionBtn.click();
```

**Review by jxn-30 — CHANGES_REQUESTED** [review]

File: src/modules/hotkeys/assets/commands/building.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                        }
+                    } else {
+                        leitstellenId = result?.leitstelle_building_id;
+                    }
+                    window.location.replace(`/buildings/${leitstellenId}`);
+                });
+        },
+        previousBuilding() {
+            const previous = this.navigationGroup?.firstElementChild;
+            (previous as HTMLElement).click();
```
[the remark this answers is not in the document]
Comment (Cr4zyc4k3) [reply src/modules/hotkeys/assets/commands/building.ts:64]: Would throw a ts-error. `TS2339: Property 'click' does not exist on type 'Element'.`

File: src/modules/hotkeys/assets/commands/building.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        validatorFunction() {
+            //Just in case. Should be in every building
+            this.navigationGroup = document.querySelector<HTMLDivElement>(
+                '#building-navigation-container'
+            );
+            return !!this.navigationGroup;
+        },
+        nextBuilding() {
+            const next = this.navigationGroup?.firstElementChild;
+            (next as HTMLElement).click();
```
[the remark this answers is not in the document]
Comment (Cr4zyc4k3) [reply src/modules/hotkeys/assets/commands/building.ts:28]: Indeed, that's a copy&paste error :see_no_evil: 

Would throw a ts-error. `TS2339: Property 'click' does not exist on type 'Element'.`

**Review by jxn-30 — APPROVED** [review]

**Pull request merged**